src/tpm2_linux.c

```diff
@@ -0,0 +1,116 @@
+/* tpm2_linux.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#include <wolftpm/tpm2_linux.h>
+#include <wolftpm/tpm2_packet.h>
+#include <wolftpm/tpm2_wrap.h> /* Needed only for WOLFTPM2_MAX_BUFFER */
+#include <sys/types.h>
```

Comment:
Please wrap this entire file in `#ifdef WOLFTPM_LINUX_DEV`. This will allow someone using a compiler that includes *.c to still build okay on a non-linux system.

Reply:
Indeed, I was thinking the same thing, what if someone build bare-metal on Linux without using /dev/tpm0 . It is added.